integration-tests/integration_test.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var server *exec.Cmd
+var kwfs *exec.Cmd
+
+func TestIntegrationTest(t *testing.T) {
+	defer cleanup()
+
+	assert := assert.New(t)
+
+	// start fake-server
+	server = exec.Command("./fake-server")
+	err := server.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	user, err := user.Current()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start keywhiz-fs
+	os.MkdirAll("./mount", 0755)
+	kwfs = exec.Command("../keywhiz-fs", "--cert", "kwfs.crt", "--key",
+		"kwfs.key", "--ca", "keywhiz.crt", "--debug", "--disable-mlock",
+		"--asuser", user.Username, "--group", lookupGroup(user.Gid),
+		"--cache-timeout", "1s", "https://localhost:8080/", "./mount")
```

Comment:
We could have the test spin up the fake server on a random port and then connect to it here. This way we don't need to build a binary and we won't hardcode a port.